merkle/src/merkle.rs

```diff
@@ -704,7 +931,9 @@ impl<T: Element, A: Algorithm<T>, K: Store<T>> MerkleTree<T, A, K> {
 
                     let chunk_nodes = {
                         // Read everything taking the lock once.
-                        let read_store = read_store_lock.read().unwrap();
+                        let read_store = read_store_lock.write().unwrap();
+                        // FIXME: Reading in the DiskStore is modyfing state at the moment (offset)
+                        //  so it can't happen in parallel with other reads/writes.
```

Comment:
How can we read an open file without affecting its state (offset)? Can we use something like https://github.com/vasi/positioned-io?

/cc @dignifiedquire

Reply:
if we use a `File` as backing store, yes that is a good solution